ql/src/java/org/apache/hadoop/hive/ql/metadata/Table.java

```diff
@@ -347,7 +351,15 @@ final public Deserializer getDeserializer(boolean skipConfError) {
 
   final public Deserializer getDeserializerFromMetaStore(boolean skipConfError) {
     try {
-      return HiveMetaStoreUtils.getDeserializer(SessionState.getSessionConf(), tTable, metaTable, skipConfError);
+      String asOfTimestamp = null;
+      if (getAsOfTimestamp() != null) {
+        ZoneId zoneId = SessionState.get() == null ? new HiveConf().getLocalTimeZone() :
+            SessionState.get().getConf().getLocalTimeZone();
+        TimestampTZ time = TimestampTZUtil.parse(PlanUtils.stripQuotes(getAsOfTimestamp()), zoneId);
+        asOfTimestamp = Long.toString(time.toEpochMilli());
```

Comment:
This calculation of Iceberg As Of information from string to long is found in multiple places, I think we should refactor this logic into a utility class of ql perhaps.